packages/shared/web/src/dashboard-view/hooks/use-leverage-farming-grid.tsx

```diff
@@ -33,35 +30,30 @@ export const useLeveragedFarmingGrid = (
   const theme = useTheme();
   const navigate = useNavigate();
   const { baseCurrency } = useAppStore();
-  const listedVaults = useAllVaults(network);
   const vaultHoldings = useVaultHoldings(network);
   const totalArbPoints = useTotalArbPoints();
   const currentSeason = useCurrentSeason();
   const [showNegativeYields, setShowNegativeYields] = useState(false);
   const [hasNegativeApy, setHasNegativeApy] = useState(false);
-  const {
-    yields: { leveragedVaults },
-    getMax,
-  } = useAllMarkets(network);
+  const currentNetworkStore = useCurrentNetworkStore();
+  const { farmingVaults, pointsVaults } =
+    currentNetworkStore.getAllListedVaultsData();
 
-  const allVaultData = listedVaults
-    .map(({ vaultAddress, name, primaryToken, vaultTVL }) => {
-      const y = getMax(
-        leveragedVaults.filter((y) => y.token.vaultAddress === vaultAddress)
-      );
+  const yieldData =
+    currentVaultType === VAULT_TYPES.LEVERAGED_POINTS_FARMING
+      ? pointsVaults
+      : farmingVaults;
+
+  const allVaultData = yieldData
+    .map(({ vaultData, debtToken, underlying, tvl, apy, token }) => {
+      // TODO: Replace with new MOBX account data when ready
       const profile = vaultHoldings.find(
-        (p) => p.vault.vaultAddress === vaultAddress
+        (p) => p.vault.vaultAddress === vaultData?.vaultAddress
```

Comment:
Based on the types above, the vaultAddress key doesnt exist. The vaultAddress key is on the token object.